tests/phpunit/tests/option/wpDetermineOptionAutoloadValue.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/**
+ * @group option
+ *
+ * @covers ::wp_determine_option_autoload_value
+ */
+class Tests_WP_Determine_Option_Autoload_Value extends WP_UnitTestCase {
+	public function set_up() {
+		add_filter( 'wp_max_autoloaded_option_size', array( $this, 'filter_max_option_size' ) );
+		parent::set_up();
+	}
+
+	/**
+	 * @ticket 42441
+	 *
+	 * @dataProvider data_values
+	 *
+	 * @param $autoload
+	 * @param $expected
+	 */
+	public function test_get_autoload_value_values( $autoload, $expected ) {
+		$test = wp_determine_option_autoload_value( null, '', '', $autoload );
+		$this->assertSame( $expected, $test );
+	}
+
+	public function data_values() {
+		return array(
+			'yes'   => array(
+				'autoload' => 'yes',
+				'expected' => 'on',
+			),
+			'true'  => array(
+				'autoload' => true,
+				'expected' => 'on',
+			),
+			'no'    => array(
+				'autoload' => 'no',
+				'expected' => 'off',
+			),
+			'false' => array(
+				'autoload' => false,
+				'expected' => 'off',
+			),
+			'null'  => array(
+				'autoload' => null,
+				'expected' => 'auto',
+			),
+		);
```

Comment:
Similarly to the above `add_option()` test, we should also add a few test cases here for invalid values, which for backward compatibility need to come back as `'on'`. Also probably we should have test cases where `'on'` and `'off'` are passed to the function.

Reply:
done